common/utils.c

```diff
@@ -428,13 +428,24 @@ TDNFYesOrNo(
 
     if(!pArgs->nAssumeYes && !pArgs->nAssumeNo)
     {
-        pr_crit("%s", pszQuestion);
-        while ((opt = getchar()) == '\n' || opt == '\r');
-        opt = tolower(opt);
-        if (opt != 'y' && opt != 'n')
-        {
-            dwError = ERROR_TDNF_INVALID_INPUT;
-            BAIL_ON_TDNF_ERROR(dwError);
+        while(1) {
+            pr_crit("%s", pszQuestion);
+            char buf[256];
```

Comment:
buf[256] = {}; instruction in line 442 may not be the case always

Reply:
I can do that, but I fail to see where this has an effect, unless `fgets()` is totally misbehaving.